src/uct/ib/rc/base/rc_iface.c

```diff
@@ -556,6 +565,16 @@ UCS_CLASS_INIT_FUNC(uct_rc_iface_t, uct_rc_iface_ops_t *ops, uct_md_h md,
     self->config.tx_cq_len          = init_attr->tx_cq_len;
 #endif
 
+    self->config.max_get_zcopy = uct_ib_iface_port_attr(&self->super)->max_msg_sz;
+    if (config->tx.max_get_zcopy != UCS_MEMUNITS_AUTO) {
+        if (config->tx.max_get_zcopy > self->config.max_get_zcopy) {
+            ucs_warn("can't set max_get_zcopy to %ld, trimmed by %ld",
+                     config->tx.max_get_zcopy, self->config.max_get_zcopy);
+        } else {
+            self->config.max_get_zcopy = config->tx.max_get_zcopy;
+        }
+    }
+
```

Comment:
1. seems weird to assign self->config.max_get_zcopy=max_msg_sz and then override it
2. i would just use ucs_min(uct_ib_iface_port_attr(&self->super)->max_msg_sz, config->tx.max_get_zcopy). so we can set the default max_get_zcopy to "inf"

Reply:
default == inf would look confusing, because looks like any message size can be handled/transferred in one op.